k8s_test_harness/util/docker_util.py

```diff
@@ -180,3 +181,28 @@ def get_image_version(image):
         text=True,
     )
     return process.stdout.strip()
+
+
+def run_entrypoint_and_assert(
+    image, entrypoint, expect_stdout_contains=None, expect_stderr_contains=None
+):
```

Comment:
Love me some typing.  You'll need to add `Union` and `Optional`

```suggestion
def run_entrypoint_and_assert(
    image: str,
    entrypoint: Union[str, List[str]],
    expect_stdout_contains: Optional[str]=None,
    expect_stderr_contains: Optional[str]=None
):
```